core/storage/activity/gae_models.py

```diff
@@ -0,0 +1,55 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for activity references."""
+
+import core.storage.base_model.gae_models as base_models
+
+from google.appengine.ext import ndb
+
+
+# The datastore model ID for the list of featured activity references. This
+# value should not be changed.
+ACTIVITY_REFERENCE_LIST_FEATURED = 'featured'
+ALL_ACTIVITY_REFERENCE_LISTS = [ACTIVITY_REFERENCE_LIST_FEATURED]
+
+
+class ActivityReferencesModel(base_models.BaseModel):
+    """Storage model for a list of activity references.
+
+    The id of each model instance is the name of the list. This should be one
+    of the constants in ALL_ACTIVITY_REFERENCE_LISTS.
+    """
+    # The types and ids of activities to show in the library page. Each item
+    # in this list is a dict with two keys: 'type' and 'id'.
+    activity_references = ndb.JsonProperty(repeated=True)
+
+    @classmethod
+    def get(cls, list_name):
```

Comment:
Well -- this is actually a get() every single time except the first time the /library page is loaded. This class is actually a bit different from most others, since its ids are fixed and drawn from a small set.

Having said that, I take your point about the inconsistency and the side-effects. I'll move the "or create" part to the services layer instead, and keep this getter pure.

One question: are you OK with the getter checking if the list_name is valid?